amptorch/preprocessing/atoms_to_data.py

```diff
@@ -38,8 +38,18 @@ def convert(
         atoms,
         idx,
     ):
+        descriptor_calculator = DescriptorCalculator(
+            images=[atoms],
+            descriptor=self.descriptor,
+            calc_derivatives=self.fprimes,
+            save_fps=self.save_fps,
+            cores=self.cores,
+            verbose=False,
+        )
+        self.descriptor_data = descriptor_calculator.prepare_descriptors()
+
```

Comment:
This instantiates the descriptor calculator each time an atoms object is provided, whereas we can build the class once for a list of atoms object instead. Did the LMDB datasets require this to be moved here, if so do you remember why?